app/models/user_group.rb

```diff
@@ -19,8 +19,12 @@ class UserGroup < ApplicationRecord
   has_many :direct_child_groups, class_name: "UserGroup", inverse_of: :parent_group, foreign_key: "parent_group_id"
   has_many :roles, foreign_key: "group_id", class_name: "UserRole"
   has_many :active_roles, -> { active }, foreign_key: "group_id", class_name: "UserRole"
+  has_many :roles_of_direct_child_groups, through: :direct_child_groups, source: :roles
+  has_many :active_roles_of_direct_child_groups, -> { active }, through: :direct_child_groups, source: :roles
   has_many :users, through: :roles
-  has_many :active_users, through: :active_roles, source: :user, class_name: "User"
+  has_many :active_users, through: :active_roles, source: :user
+  has_many :users_of_direct_child_groups, through: :roles_of_direct_child_groups, source: :user
+  has_many :active_users_of_direct_child_groups, through: :active_roles_of_direct_child_groups, source: :user
```

Comment:
See above for alternative naming suggestions.

Reply:
Done.